mlir-tensorrt/compiler/include/mlir-tensorrt/Compiler/Client.h

```diff
@@ -145,8 +172,43 @@ class CompilerClient {
   /// used to create the PM.
   llvm::DenseMap<PassManagerKey, std::unique_ptr<CompilationTaskBase>>
       cachedPassManagers;
+
+  static llvm::ManagedStatic<llvm::StringMap<OptionsConstructorFuncT>> registry;
```

Comment:
This shouldn't be exposed in the header. It can be a global static variable in the C++ file.